.github/workflows/publish.yaml

```diff
@@ -50,17 +51,31 @@ jobs:
             "$GITHUB_SERVER_URL/$GITHUB_REPOSITORY/actions/runs/$GITHUB_RUN_ID" > ./version.json
             # Show complete version.json for debugging
             cat ./version.json
-      - name: Extract metadata (tags, labels) for Docker
-        id: meta
+      - name: Extract metadata for Google Artifact Registry
+        id: metagar
         uses: docker/metadata-action@v5
         with:
+          flavor:
+            # don't automatically tag with `latest`; we do this conditionally in the `tags` section
+            latest=false
           images: |
-            ${{ env.DOCKERHUB_IMAGE_NAME }}
             ${{ env.GAR_LOCATION }}-docker.pkg.dev/${{ env.GCP_PROJECT_ID }}/${{ env.GAR_REPOSITORY }}/${{ env.GAR_IMAGE_NAME }}
-          # https://github.com/marketplace/actions/docker-metadata-action#tags-input
           tags: |
-            type=raw,value=latest
-            type=raw,value=${{ env.LATEST_TAG }}
+            type=raw,value=${{ env.LATEST_TAG }},enable=${{ github.event_name == 'push' }}
+            type=sha,format=long,enable=${{ github.event_name == 'push' }}
+            type=semver,pattern={{raw}},enable=${{ github.event_name == 'release' }}
+      - name: Docker Metadata for Docker Hub
```

Comment:
Do we still need to push to Docker Hub? For crash ingestion, we are only using GAR now. It's still possible to make the GAR images public if desired.